### PR TITLE
Endpoint for smoothing data #612

### LINKED ISSUE #612: Endpoint for smoothing data
subtask for #606 

add endpoint that return retrieved smoothed data from analytic

### PR DESCRIPTION
**Pull request by amper43**
fixes #612 

changes:
- added ability to get smoothed data from HSR
- added new task type for data processing

### CODE REVIEW

File: analytics/analytics/detectors/anomaly_detector.py [hunk 1]
```diff
@@ -71,8 +85,27 @@ def get_window_size(self, cache: Optional[ModelCache]) -> int:
...
+            segments = [{'from': segment[0], 'to': segment[1]} for segment in segments]
+
+            united_result.segments.extend(segments)
+            united_result.cache = result.cache
+            united_result.last_detection_time = result.last_detection_time
+
+        return united_result
+
+    def process_data(self, data: pd.DataFrame, cache: ModelCache) -> ProcessingResult:
+        smoothed = utils.exponential_smoothing(data, cache['alpha'], cache.get('lastValue'))
```
Comment (VargBurz) [inline analytics/analytics/detectors/anomaly_detector.py]: if `data: pd.DataFrame` -> 
smoothed = utils.exponential_smoothing(data['value'], cache['alpha'], cache.get('lastValue'))

File: analytics/analytics/detectors/anomaly_detector.py [hunk 2]
```diff
@@ -71,8 +85,27 @@ def get_window_size(self, cache: Optional[ModelCache]) -> int:
...
+        result = ProcessingResult(smoothed.values.tolist())
```
Comment (VargBurz) [inline analytics/analytics/detectors/anomaly_detector.py]: res = []
for idx, val in enumerate(smoothed.values):
    res.append([data['timestamp'][idx], val])
smoothed = res

File: analytics/analytics/analytic_unit_worker.py [hunk 3]
```diff
@@ -1,13 +1,14 @@
 import config
-import detectors
+import  detectors
```
Comment (jonyrock) [inline analytics/analytics/analytic_unit_worker.py]:
```suggestion
import detectors
```

File: server/src/models/analytic_units/anomaly_analytic_unit_model.ts [hunk 4]
```diff
@@ -0,0 +1,85 @@
+import { AnalyticUnit } from './analytic_unit_model';
+import { AnalyticUnitId, AnalyticUnitStatus, DetectorType } from './types';
+
+import { Metric } from 'grafana-datasource-kit';
+
+export class AnomalyAnalyticUnit extends AnalyticUnit {
```
Comment (jonyrock) [inline server/src/models/analytic_units/anomaly_analytic_unit_model.ts]: why does this class appear in PR with title "Endpoint for smoothing data" ? 
It makes no sense to me

**Comment by jonyrock** [comment]
I decline this PR: 
1. changes in description don't correspond to changes in code. 
2. If this task about endpoint, then you can't add any new functionality to `do_detect` or whatever

**Pull request closed without merging**